SwitchBindingInstance.groovy

```diff
@@ -267,7 +267,15 @@ def syncSwitchState(triggeredDeviceId, onOrOff) {
 			if (onOrOff) {
 				log "BINDING: ${s.displayName} -> on()"
                 if (s.currentValue('switch', true) != 'on') {
-                    s.on()
+                    if (s.getSetting("transitionTime") != null && s.hasAttribute('level')) {
```

Comment:
So this doesn't require any configuration of the SwitchBinding itself?  It just reads if the device driver has a transitionTime setting?

Reply:
That's right. If the device being turned on has the transitionTime setting, then the app will honor that setting. I didn't see any need for the app to make the user specify it, since the user will have already specified the desired behavior via the device's transitionTime setting.